Assets/Scripts/PlayerIcon.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.UI;
+
+/// <summary>
+/// Manages Player Icon Location
+/// Appears when camera switch to map mode and when in isometric view
+/// 
+/// </summary>
+public class PlayerIcon : MonoBehaviour
+{
+    public Transform player;
+    private static bool _isoActive;
+    private static bool _mapActive;
+    private bool _onSlingShot;
+
+    private void Start()
+    {
+        
+        CameraSwitch.OnMapSwitch += SetMapActive;
+        IsometricStarView.OnIsometricStarView += SetIsometricActive;
+        Orbit.OnSlingShot += OnSlingShot;
+        GetComponent<Image>().enabled = false;
+
+    }
+
+    private void OnSlingShot(bool onSlingShot, Transform _starToGo)
+    {
+        _onSlingShot = onSlingShot;
+        _isoActive = false;
+        GetComponent<Image>().enabled = false;
+    }
+
+    private void FixedUpdate()
+    {
+        transform.position = player.position;
+    }
+
+    private void SetMapActive(bool mapActive)
+    {
+        _mapActive = mapActive;
+
+        HandleImageEnable();
+
+
+    }
+    
+    private void SetIsometricActive(bool isoActive)
+    {
+        _isoActive = isoActive;
+        
+        Invoke(nameof(HandleImageEnable), 4.0f);
```

Comment:
Why is the camera pan here listed as 4 seconds? Earlier it was listed at 0.5 seconds.

Furthermore, if the camera pan only takes 0.5 seconds, then we don't need to delay the appearance of the player icon at all.